Documentation/bootstrapper.md

```diff
@@ -0,0 +1,48 @@
+# tectonic-torcx-bootstrap
+
+This document gives a brief overview on how the bootstrapper works. Logic described here is specific to bootstrap context, actual code is generic to also fit as a CLUO daemonset hook.
+
+## Execution and context
+
+The bootstrapper runs on every new Tectonic node, at most once, before the node can join the cluster.
+It must write the `/etc/kubernetes/kubelet.env` in order for the kubelet to start.
+It is deployed via Ignition as a systemd service called `k8s-node-bootstrap`.
+This service is installed and owned by tectonic-installer via the relevant [service unit][bootstrap-service].
+
+## Configuration knobs
+
+A few configuration options are available on the command-line. These are the most interesting:
+ * `--upgrade-os=<bool>`: whether to check for new OS updates. Defaults to `true`
+ * `--torcx-skip-setup=<bool>`: whether to skip all torcx-related steps. Defaults to `false`
+ * `--no-verify-signatures=<bool>`: skip GPG verification on addons manifest. Default to `false`
+ * `--force-kube-version=<string>`: force a specific kubernetes version, skipping default autodetection logic
+ * `--torcx-manifest-url=<string>`: URL template for torcx addons manifest. More details below
+
+Currently, torcx addons manifests are available at the following URL template:
+```
+https://tectonic-torcx.release.core-os.net/manifests/{{.Board}}/{{.OSVersion}}/torcx_manifest.json
+```
+Template variables are replaced with node-specific values. A detached signature is provided at the same URL suffixed with a `.asc` extension.
+
+## Sources of information
+
+Bootstrapper tries to gather state from the cluster and from a [remote bucket][remote], in order to prepare an up-to-date Kubernetes node.
+
+This is the logic flow and the information sources it queries:
+ 1. Get cluster version from api-server `/version` endpoint
+ 1. Iff previous step permanently failed, use `/etc/kubernetes/installer/kubelet.env` to determine original cluster version (from installation time)
```

Comment:
I would add some kind of explicit description that this is needed for bootstrap, since there will be no api server.

Reply:
Yep. I'll add few sentences to better describe node bootstrap in cluster bootstrap vs existing cluster scenario.

Reply:
Added at the bottom of this numbered list.